rs/tests/execution/general_execution_tests/nns_shielding.rs

```diff
@@ -141,6 +142,103 @@ pub fn mint_cycles_not_supported_on_application_subnet(env: TestEnv) {
     });
 }
 
+pub fn mint_cycles128_supported_only_on_cycles_minting_canister(env: TestEnv) {
+    let nns_node = env.get_first_healthy_nns_node_snapshot();
+    let specified_id = nns_node.get_last_canister_id_in_allocation_ranges();
+    // Check that 'specified_id' is not 'CYCLES_MINTING_CANISTER_ID'.
+    assert_ne!(specified_id, CYCLES_MINTING_CANISTER_ID.into());
+    let nns_agent = nns_node.build_default_agent();
+    block_on(async move {
+        let not_cmc = UniversalCanister::new_with_cycles(&nns_agent, specified_id, *INITIAL_CYCLES)
+            .await
+            .unwrap()
+            .canister_id();
+
+        let before_balance = get_balance(&not_cmc, &nns_agent).await;
+        assert_balance_equals(
+            *INITIAL_CYCLES,
+            Cycles::from(before_balance),
+            BALANCE_EPSILON,
+        );
+
+        let res = nns_agent
+            .update(&not_cmc, "update")
+            .with_arg(
+                wasm()
+                    .mint_cycles128(Cycles::from((1u128 << 64) + 2u128))
+                    .reply_data_append()
+                    .reply()
+                    .build(),
+            )
+            .call_and_wait()
+            .await
+            .expect_err("should not succeed");
+
+        assert_eq!(
+            res,
+            AgentError::CertifiedReject(
+                RejectResponse {
+                    reject_code: RejectCode::CanisterError,
+                    reject_message: format!(
+                        "Error from Canister {}: Canister violated contract: ic0.mint_cycles cannot be executed on non Cycles Minting Canister: {} != {}.\nThis is likely an error with the compiler/CDK toolchain being used to build the canister. Please report the error to IC devs on the forum: https://forum.dfinity.org and include which language/CDK was used to create the canister.",
```

Comment:
mint_cycles -> mint_cycles128

Reply:
No, because the source of the message is the `CyclesAccountManager`, which deals in Cycles (not primitive wasm types), so it is agnostic with respect to which system API endpoint was called. 

But these discrepancies will go when we remove the old mint_cycles endpoint, after the CMC has been upgraded.